services.html

```diff
@@ -0,0 +1,153 @@
+			<hr>
+			Ordering
+			<ul>
+				<li><a href="fam_barrier_all.html">fam_barrier_all</a></li>
+				<li><a href="fam_fence.html">fam_fence</a></li>
+				<li><a href="fam_quiet.html">fam_quiet</a></li>
+			</ul>
+			<hr>
+		</nav>
+		<article>
+		<h1>OpenFAM 2.0 Services</h1>
+<p> The initial reference implementation of OpenFAM was built using a single memory server.
+ As expected, the memory server becomes a bottleneck even for a small number of concurrent PEs.
+For OpenFAM version 2.0, we have made significant changes to the design of the memory server,
+with these high level requirements.
+
+<ul>
+<li>Support for larger pool of memory by scaling to multiple memory servers.
+Support larger region sizes that exceed the size of a single node thereby making large pool of
+memory accessible to the application.</li>
+<li>Because regions are intended to be large containers for data items, to improve data path performance,
+we need to allow the same region to encompass multiple memory nodes. This would reduce both network bottlenecks
+at the network interfaces, as well as performance bottlenecks due to limited compute capacity at a given memory server.</li>
+<li>We need to consider dynamic addition of memory servers when additional memory is required, without
+disturbing previously allocated memory.</li>
+</ul>
+<figure>
+<img src="images/openfam2.0memoryserver.png" width="450px">
+<figcaption>Figure 1: OpenFAM 2.0 Memory server components</figcaption>
+</figure>
+<p> There are three main components in the design.
+<ol>
+<li>The client interface service (CIS): All the PE’s talk to the CIS for memory server operations.</li>
+<li>The metadata management service: Region and data item metadata will be hosted in the metadata management service.
+The service could either run in same node as a CIS or a memory server, or it can run in dedicated metadata management nodes.</li>
+<li>The memory management service: The actual memory will be served from memory servers.
+For data path (get,put and atomic) operations PEs will talk to the memory servers directly.
+For memory management and metadata operations, like allocate, lookup, create_region, etc.,
+the CIS will communicate with memory servers via an RPC service or via direct functions calls.</li>
+</ol>
+We have defined an RPC path and a direct path for all the three components – CIS, memory management service and metadata management service.
+Yaml based configuration parameters are available to decide how these services are running.  If the components are running as separate services,
+a grpc call is made to the service. If the services are running within other component (direct path), a function call is made.
+<p> Now let us look at some of the possible combinations.
+<ol>
+<li>All services running separately:
+If all the components are running as separate services, a grpc call will be made to the services.
+As shown in Figure 1, PE will make an RPC call to CIS, in-turn CIS will make an RPC call to metadata
```

Comment:
Figure 1 does not show any RPC calls being made. Again No future tense for completed work.